shared/warning.ts

```diff
@@ -0,0 +1,124 @@
+import { enumFromStringValue } from './utils';
+
+export enum WarningType {
+  Unknown = 'Unknown',
+  WriteFilePermission = 'WriteFilePermission',
+}
+
+export enum WriteFilePermissionWarningKinds {
```

Comment:
```suggestion
export enum WriteFilePermissionWarningType {
```
to keep it consistent with WarningType

Reply:
Why it should be consistent with `WarningType`?
It is completely two different types, which do not intersect.
So I choose a different word specifically to clearly distinguish them and avoid confusing with `WarningType`.

However, better to rename it `Kinds` -> `Kind` ;)